dbt_platform_helper/domain/versioning.py

```diff
@@ -55,34 +55,59 @@ def __init__(
         environment_variable_provider: EnvironmentVariableProvider = EnvironmentVariableProvider(),
         latest_version_provider: VersionProvider = PyPiLatestVersionProvider,
         installed_version_provider: InstalledVersionProvider = InstalledVersionProvider(),
-        skip_versioning_checks: bool = None,
+        allow_override_of_versioning_checks: bool = None,
         platform_helper_version_override: str = None,
     ):
         self.io = io
         self.config_provider = config_provider
         self.latest_version_provider = latest_version_provider
         self.installed_version_provider = installed_version_provider
-        self.skip_versioning_checks = (
-            skip_versioning_checks if skip_versioning_checks is not None else skip_version_checks()
+        self.allow_override_of_versioning_checks = (
+            allow_override_of_versioning_checks
+            if allow_override_of_versioning_checks is not None
+            else allow_override_of_versioning_checks_fn()
         )
         self.environment_variable_provider = environment_variable_provider
         self.platform_helper_version_override = platform_helper_version_override
 
-    def is_managed(self):
+    def is_auto(self):
         platform_config = self.config_provider.load_unvalidated_config_file()
         default_version = platform_config.get("default_versions", {}).get("platform-helper")
         return default_version == "auto"
 
-    def get_required_version(self):
-        platform_config = self.config_provider.load_unvalidated_config_file()
-        required_version = platform_config.get("default_versions", {}).get("platform-helper")
-        self.io.info(required_version)
-        return required_version
+    def get_required_version(self) -> str:
+        if self.is_auto():
+            return str(self.get_version_status().latest)
+        else:
+            return self.get_default_version()
 
-    # Used in the generate command
-    def check_platform_helper_version_mismatch(self):
-        if self.skip_versioning_checks:
+    def _check_auto_environment(self):
```

Comment:
Could we rename this `check_configured_for_auto_environment` or similar for greater clarity?